packages/core/src/plugins/ConsentPlugin.ts

```diff
@@ -84,14 +84,14 @@ export class ConsentPlugin extends Plugin {
           const preferences = event.context?.consent?.categoryPreferences || {};
 
           if (plugin.key === SEGMENT_DESTINATION_KEY) {
+            const noneConsented = Object.values(preferences).every(
+              (consented) => !consented
+            );
+
             return (
               this.isConsentUpdateEvent(event) ||
-              !(
-                Object.values(preferences).every((consented) => !consented) &&
-                Object.entries(settings)
-                  .filter(([k]) => k !== SEGMENT_DESTINATION_KEY)
-                  .every(([_, v]) => this.containsConsentSettings(v))
-              )
+              !this.isConsentFeatureSetup() ||
+              !(noneConsented && !this.hasUnmappedDestinations())
```

Comment:
@silesky @oscb I'm tunnel visioned, can you folks help me simplify line 94? I'm sure there's a way to not have these nested negations, just can't figure how??

Note that if expression on line 91 resolves to `true`, then that means event is allowed to flow, if `false` it is blocked.

Reply:
I think the `isConsentFeatureSetup` is an interesting one to have here. Cause potentially it might throw a "false" positive and let all events flow if the settings object is not loaded (f.e. you have bad connection). I guess this is more of an scenario where we should let the client control what they want to do in that scenario. Not sure really if this has been talked in Consent Management before. Anyway. We don't have to solve this yet.

I think the line `!(noneConsented && !this.hasUnmappedDestinations())` could be rewritten as `!noneConsented  || !this.hasUnmappedDestinations()`. If I'm understanding correctly the spec you want to let things flow if everything is consented or when there's at least an unmapped destination.

Reply:
another idea to avoid many negations which make it harder to understand is to swap `noneConsented` to `allConsented` so that you don't have to negate.